spv/merkle_path_validation.go

```diff
@@ -0,0 +1,84 @@
+package spv
+
+import (
+	"context"
+	"errors"
+
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/libsv/go-bt/v2"
+)
+
+func verifyMerkleRoots(dBeef *paymail.DecodedBEEF, provider MerkleRootVerifier) error {
+	if err := ensureInputTransactionsArePresentInBump(dBeef); err != nil {
```

Comment:
Just thinking out loud:
If we will do an SPV for each parent transaction. Then maybe we could do it in a simple loop like:
1. take next transaction
2. does it have parent
   a. No -> add merkle root to verify
   b. Yes -> make SPV on it without merkle root verification
3. Repeat for each transaction in the list
4. Verify collected merkle roots.

Reply:
done